packages/editor/src/components/widgets/tag/Tags.tsx

```diff
@@ -1,80 +1,165 @@
+            aria-label='Add new tag'
             disabled={editorContext.readonly}
           >
             <IvyIcon icon={IvyIcons.Close} />
+            <input
+              disabled={editorContext.readonly}
+              className='new-tag-input'
+              {...getInputProps({
+                onFocus: () => {
+                  openMenu();
+                  setAddValue('');
+                },
+                ref: inputRef,
+                value: addValue,
+                'aria-label': 'New Tag'
+              })}
+              style={{ width: inputValue.length * 8 > 28 ? inputValue.length * 8 : 28 }}
+            />
           </button>
-          <span>{tag}</span>
+          <ul {...getMenuProps()} className='combobox-menu'>
+            {isOpen &&
+              dropDownTags.map((item, index) => (
+                <li
+                  className={`combobox-menu-entry ${highlightedIndex === index ? 'hover' : ''} ${selectedItem === item ? 'selected' : ''}`}
+                  key={`${item}${index}`}
+                  {...getItemProps({ item, index })}
+                >
+                  <span>{item}</span>
+                </li>
+              ))}
+          </ul>
         </div>
-      ))}
-      <Popover.Root open={isOpen} onOpenChange={handleAddPopoverChange}>
-        <Popover.Trigger asChild>
-          <button className='tag tag-add' aria-label='Add new tag' disabled={editorContext.readonly}>
-            <IvyIcon icon={IvyIcons.Plus} />
-            <span>Add</span>
-          </button>
-        </Popover.Trigger>
-        <Popover.Portal container={editorContext.editorRef.current}>
-          <Popover.Content className='popover-content' sideOffset={5} collisionBoundary={editorContext.editorRef.current}>
-            <Fieldset label='New Tag' {...tagFieldset.labelProps}>
-              <Input value={newTag} {...keyboardProps} onChange={change => setNewTag(change)} {...tagFieldset.inputProps} />
-            </Fieldset>
-            <Popover.Close className='popover-close' aria-label='Close'>
+        {props.tags.map((tag, index) => (
+          <div key={`${tag}-${index}`} className='added-tag' role='gridcell'>
+            <button
+              className='tag-remove'
+              onClick={() => {
+                removeTag(tag);
+              }}
+              aria-label={`Remove Tag ${tag}`}
+              {...keyboardProps}
+              disabled={editorContext.readonly}
+            >
               <IvyIcon icon={IvyIcons.Close} />
-            </Popover.Close>
-            <Popover.Arrow className='popover-arrow' />
-          </Popover.Content>
-        </Popover.Portal>
-      </Popover.Root>
-    </div>
+            </button>
+            <span>{tag}</span>
+          </div>
+        ))}
+      </div>
+      {hasSpaceInTag && (
+        <MessageText
+          message={{ message: 'Spaces are not supported inside the tag. Tags with spaces will not be saved.', severity: 'WARNING' }}
```

Comment:
ah wasn't aware of this...
Instead of this big hasSpaceInTag logic, maybe we should just remove all spaces from the input. When the user adds it then the new tag will appear without spaces. I think this should be enough.

Reply:
I would remove this hint. We can later add it again if it is really needed.